proof_of_stake/src/lib.rs

```diff
@@ -805,14 +807,28 @@ pub trait PosBase {
                 if let (Some(prev_epoch), Some(prev_validators)) =
                     (previous_epoch, prev_validators)
                 {
-                    if prev_validators.active.contains(validator) {
+                    let prev_validator_stake =
+                        self.validator_stake(&validator.address, prev_epoch);
+                    if prev_validators.active.contains(validator)
+                        || into_tm_voting_power(
```

Comment:
we should calculate `into_tm_voting_power(params.tm_votes_per_token, validator.bonded_stake)` once, store it in a variable, and reuse it in all these comparisons

Reply:
& ditto for previous validator stake